compiler/z/env/OMRCPU.cpp

```diff
@@ -112,7 +112,56 @@ OMR::Z::CPU::CPU()
    :
    OMR::CPU(),
    _supportedArch(z9)
-   {}
+   {
+   if (self()->isAtLeast(OMR_PROCESSOR_S390_Z10) && TR::Options::getCmdLineOptions()->getOption(TR_DisableZ10))
+      self()->setProcessor(OMR_PROCESSOR_S390_FIRST);
```

Comment:
Using `self()` within a constructor relies on undefined behavior [1]. Is it possible to avoid using it?

[1] https://github.com/eclipse/omr/pull/4798#discussion_r382624775